plugin/evm/atomic/state/atomic_trie_test.go

```diff
@@ -904,9 +819,76 @@ func benchmarkApplyToSharedMemory(b *testing.B, disk database.Database, blocks u
 	b.ReportAllocs()
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		backend.(*atomicBackend).sharedMemory = testSharedMemory()
+		backend.sharedMemory = atomictest.TestSharedMemory()
 		assert.NoError(b, backend.MarkApplyToSharedMemoryCursor(0))
 		assert.NoError(b, db.Commit())
 		assert.NoError(b, backend.ApplyToSharedMemory(lastAcceptedHeight))
 	}
 }
+
+// verifyOperations creates an iterator over the atomicTrie at [rootHash] and verifies that the all of the operations in the trie in the interval [from, to] are identical to
+// the atomic operations contained in [operationsMap] on the same interval.
+func verifyOperations(t testing.TB, atomicTrie *AtomicTrie, codec codec.Manager, rootHash common.Hash, from, to uint64, operationsMap map[uint64]map[ids.ID]*avalancheatomic.Requests) {
+	t.Helper()
+
+	// Start the iterator at [from]
+	fromBytes := make([]byte, wrappers.LongLen)
+	binary.BigEndian.PutUint64(fromBytes, from)
+	iter, err := atomicTrie.Iterator(rootHash, fromBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Generate map of the marshalled atomic operations on the interval [from, to]
+	// based on [operationsMap].
+	marshalledOperationsMap := make(map[uint64]map[ids.ID][]byte)
```

Comment:
nit rename to `marshalledHeightToBlockRequests`? A bit long though so feel free to ignore